server/src/main/java/org/apache/druid/server/http/MetadataResource.java

```diff
@@ -471,4 +471,23 @@ public Response getDataSourceInformation(
     );
     return Response.status(Response.Status.OK).entity(authorizedDataSourceInformation).build();
   }
+
+  /**
+   * @return all bootstrap segments determined by the coordinator.
+   */
+  @POST
```

Comment:
Why POST? Is it to accomodate future possible changes to this API to fetch all segments required to bootstrap a historical? For that, I think we would at most need a couple of flags which could be GET query params.

Reply:
Yep, see this https://github.com/apache/druid/pull/16609#discussion_r1649083206. I will adjust the API semantics to tailor them to this PR's needs.